src/main/java/com/openshift/internal/restclient/model/Pod.java

```diff
@@ -61,10 +76,35 @@ public String getHost() {
 		}
 		return images;
 	}
+	
+	/**
+	 * The logic of the method is a copied from 'podStatus' function of
+	 * [app/scripts/filters/resources.js] of [openshift/origin-web-console]
+	 */
 
 	@Override
 	public String getStatus() {
-		return asString(POD_STATUS);
+		if (has(POD_DELETION_TIMESTAMP)) {
+			return "Terminating";
+		}
+		ModelNode node = get(POD_STATUS_CONTAINER_STATUSES);
+		if (node.getType() == ModelType.LIST) {
+			for (ModelNode containerStatus : node.asList()) {
+				for (String containerReason: POD_STATUS_CONTAINER_STATES.keySet()) {
```

Comment:
this (l93 - l98) could be extracted to a method #getPodContainerState for improved readability